### PR TITLE
fix(panel, flow-item): prevent footer slots from conflicting with each other

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #9855

## Summary

- prevent footer slots from conflicting
- modify stories
- add doc

### CODE REVIEW

**Comment by driskull** [comment]
@geospatialem can you review doc changes?

File: packages/calcite-components/src/components/flow-item/flow-item.tsx [hunk 1]
```diff
@@ -52,10 +52,10 @@ import { CSS, ICONS, SLOTS } from "./resources";
  * @slot header-content - A slot for adding custom content to the component's header.
  * @slot header-menu-actions - A slot for adding an overflow menu with `calcite-action`s inside a `calcite-dropdown`.
  * @slot fab - A slot for adding a `calcite-fab` (floating action button) to perform an action.
- * @slot footer - A slot for adding custom content to the component's footer.
+ * @slot footer - A slot for adding custom content to the component's footer. Should not be used in conjunction with footer-start or footer-end slots.
```
Comment (geospatialem) [inline packages/calcite-components/src/components/flow-item/flow-item.tsx:55]: WDYT about simplifying the language a bit? Also would add backticks and double quotes to the slot names:

```suggestion
 * @slot footer - A slot for adding custom content to the component's footer. Should not be used with the `"footer-start"` or `"footer-end"` slots.
```

File: packages/calcite-components/src/components/flow-item/flow-item.tsx [hunk 2]
```diff
@@ -52,10 +52,10 @@ import { CSS, ICONS, SLOTS } from "./resources";
...
  * @slot footer-actions - [Deprecated] Use the `"footer"` slot instead. A slot for adding `calcite-button`s to the component's footer.
- * @slot footer-end - A slot for adding a trailing footer custom content.
- * @slot footer-start - A slot for adding a leading footer custom content.
+ * @slot footer-end - A slot for adding a trailing footer custom content. Should not be used in conjunction with footer slot.
+ * @slot footer-start - A slot for adding a leading footer custom content. Should not be used in conjunction with footer slot.
```
Comment (geospatialem) [inline packages/calcite-components/src/components/flow-item/flow-item.tsx:58]: Similar pattern as above:

```suggestion
 * @slot footer-end - A slot for adding a trailing footer custom content. Should not be used with the `"footer"` slot.
 * @slot footer-start - A slot for adding a leading footer custom content. Should not be used with the `"footer"` slot.
```

File: packages/calcite-components/src/components/panel/panel.tsx [hunk 3]
```diff
@@ -58,10 +58,10 @@ import { CSS, ICONS, SLOTS } from "./resources";
  * @slot header-content - A slot for adding custom content to the header.
  * @slot header-menu-actions - A slot for adding an overflow menu with actions inside a `calcite-dropdown`.
  * @slot fab - A slot for adding a `calcite-fab` (floating action button) to perform an action.
- * @slot footer - A slot for adding custom content to the component's footer.
+ * @slot footer - A slot for adding custom content to the component's footer. Should not be used in conjunction with footer-start or footer-end slots.
  * @slot footer-actions - [Deprecated] Use the `footer-start` and `footer-end` slots instead. A slot for adding `calcite-button`s to the component's footer.
- * @slot footer-end - A slot for adding a trailing footer custom content.
- * @slot footer-start - A slot for adding a leading footer custom content.
+ * @slot footer-end - A slot for adding a trailing footer custom content. Should not be used in conjunction with footer slot.
+ * @slot footer-start - A slot for adding a leading footer custom content. Should not be used in conjunction with footer slot.
```
Comment (geospatialem) [inline packages/calcite-components/src/components/panel/panel.tsx:64]: Same as above, if you agree with the updates.

**Review by geospatialem — COMMENTED** [review]
Doc considerations to simplify the descriptions.

**Review by benelan — APPROVED** [review]
:+1:

**Pull request merged**